### PR TITLE
Update pre commit

### PR DESCRIPTION
**Pull request by j042**
update/add pre-commit hooks and changes to pass those new ones

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -43,8 +43,20 @@ repos:
   - id: black
     args: [--line-length=79]
 
-- repo: https://github.com/PyCQA/flake8
-  rev: 6.1.0
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.0.291
   hooks:
-  - id: flake8
-    args: [--extend-ignore=E203]
```
Comment (clemens-fricke) [inline .pre-commit-config.yaml:50]: This is now enforced? There does not seem any impact of this rule change just wondering why it was ignored before.

**Comment by j042** [comment]
did examples run fine?

File: gustaf/show.py [hunk 2]
```diff
@@ -161,8 +161,8 @@ def cam_tuple_to_list(dict_cam):
         list_of_showables = []
         for sl in show_list:
             if not isinstance(sl, list):
-                sl = [sl]
-            for k, item in enumerate(sl):
+                sl = [sl]  # noqa: PLW2901
```
Comment (clemens-fricke) [inline gustaf/show.py:164]: Otherwise error that the iterator variable is overwritten inside the loop. Here it makes sense.

File: gustaf/show.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def show(*gus_obj, **kwargs):
 
     if vis_b.startswith("vedo"):
         return show_vedo(*gus_obj, **kwargs)
-    elif vis_b.startswith("trimesh"):
+    elif vis_b.startswith("trimesh"):  # noqa: SIM114
```
Comment (clemens-fricke) [inline gustaf/show.py:52]: Otherwise, this gives an error that both elif could be handled in a single statement with an or.

**Comment by j042** [comment]
this `keys()` replacement I see there should be fine I guess as they are all for dict. otherwise would be related to #148

File: gustaf/utils/connec.py [hunk 4]
```diff
@@ -191,10 +191,7 @@ def faces_to_edges(faces):
 
     for i in range(vertices_per_face):
         # v_ind : corresponding vertex index for i value
-        if i == int(vertices_per_face - 1):
-            v_ind = 0
-        else:
-            v_ind = i + 1
+        v_ind = 0 if i == int(vertices_per_face - 1) else i + 1
```
Comment (clemens-fricke) [inline gustaf/utils/connec.py:194]: Not sure if we want this simplification. Could make it harder to read?

Should be Rule SIM108, i think.

File: gustaf/utils/connec.py [hunk 5]
```diff
@@ -238,10 +235,7 @@ def range_to_edges(range_, closed=False, continuous=True):
 
     # continuous edges
     indices = np.repeat(indices, 2)
-    if closed:
-        indices = np.append(indices, indices[0])[1:]
-    else:
-        indices = indices[1:-1]
+    indices = np.append(indices, indices[0])[1:] if closed else indices[1:-1]
```
Comment (clemens-fricke) [inline gustaf/utils/connec.py:238]: Here the same.

File: gustaf/utils/tictoc.py [hunk 6]
```diff
@@ -119,6 +119,6 @@ def summary(self, log=True, print_=False):
             if log:
                 self._logger(*message)
             if print_:
-                print(*message)
+                print(*message)  # noqa: T201
```
Comment (clemens-fricke) [inline gustaf/utils/tictoc.py:122]: print statements are forbidden following rule T201. Here we can ignore this rule due to explicitly asking for a print of the message.

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  "PL",          # pylint
+  "SIM",         # flake8-simplify
+  "T20",         # flake8-print
+  "NPY",         # numpy specific rules
+  "I",           # isort specific rules
+  "UP",          # pyupdate specific rules
+  "C400","C401","C402","C403","C404","C405" # additional pyupgrade rules
+]
+fixable = ["ALL"]
+target-version = "py37"
```
Comment (clemens-fricke) [inline pyproject.toml:17]: py36 is not available in ruff. That is why I checked if 3.6 still works with the current rules. (It does)

File: pyproject.toml
Comment (clemens-fricke) [inline pyproject.toml:1]: This configuration was mostly directly taken from `splinepy`, but adapted for gustaf.

**Review by clemens-fricke — APPROVED** [review]
Since I now made a lot of commits as well, @j042 please look into at least the comments I made to make sure these changes are ok.

**Comment by clemens-fricke** [comment]
Examples and test ran fine.

File: gustaf/show.py (hunk 2 above)
Comment (j042) [reply gustaf/show.py:164]: I guess I don't even use `k` there, right?

File: gustaf/utils/connec.py (hunk 4 above)
Comment (j042) [reply gustaf/utils/connec.py:194]: Seems fine

File: gustaf/utils/connec.py (hunk 5 above)
Comment (j042) [reply gustaf/utils/connec.py:238]: a bit weird but I am okay with leaving formatter to do its job

Comment (j042) [reply gustaf/utils/connec.py:238]: also this seems like a bit of over kill - maybe next PR

**Comment by j042** [comment]
thank you for the awesome extension

**Pull request merged**